pennylane/templates/subroutines/permute.py

```diff
@@ -157,15 +161,62 @@ def __init__(self, permutation, wires, do_queue=True, id=None):
             # Make sure everything in the permutation has an associated label in wires
         for label in permutation:
             if label not in wires:
-                raise ValueError(f"Cannot permute wire {label} not present in wire set.")
+                raise ValueError(
+                    f"Cannot permute wire {label} not present in wire set.")
 
-        self._hyperparameters = {"permutation": permutation}
+        self._hyperparameters = {"permutation": permutation, 'wires': wires}
```

Comment:
No need to add `wires` to the `self._hyperparameters` dictionary if we are using the `Permute.matrix` method (because we can use `self.wires` instead).